packages/debugger-extension/src/index.ts

```diff
@@ -367,25 +369,35 @@ const service: JupyterFrontEndPlugin<IDebugger> = {
   autoStart: true,
   provides: IDebugger,
   requires: [IDebuggerConfig],
-  optional: [INotebookTracker, IConsoleTracker, IDebuggerSources, ITranslator],
+  optional: [IDebuggerDisplayRegistry, IDebuggerSources, ITranslator],
   activate: (
     app: JupyterFrontEnd,
     config: IDebugger.IConfig,
-    notebookTracker: INotebookTracker | null,
-    consoleTracker: IConsoleTracker | null,
+    displayRegistry: DebuggerDisplayRegistry,
     debuggerSources: IDebugger.ISources | null,
     translator: ITranslator | null
   ) =>
     new Debugger.Service({
       config,
-      notebookTracker,
-      consoleTracker,
+      displayRegistry,
       debuggerSources,
       specsManager: app.serviceManager.kernelspecs,
       translator
     })
 };
 
+/**
+ * A plugin providing the debugger display registry.
+ */
+const displayRegistry: JupyterFrontEndPlugin<DebuggerDisplayRegistry> = {
```

Comment:
In general we should have an interface for things we provide like

```
interface IDebuggerDisplayRegistry {
}
```

so this would become:

```suggestion
const displayRegistry: JupyterFrontEndPlugin<IDebuggerDisplayRegistry> = {
```

and then where you define the class it would be:

```
class DebuggerDisplayRegistry implements IDebuggerDisplayRegistry {
```

Reply:
Done in [2ba22c5](https://github.com/jupyterlab/jupyterlab/pull/18043/commits/2ba22c58c8a010ecf7adb309b5db1a3592bb446e)